### PR TITLE
ROLLUP-286: add code coverage

### PR DESCRIPTION
**Pull request by antomor**
## What

- Add the code coverage tool and describe how to run it

## Why

- To keep track of the unit test coverage

### CODE REVIEW

File: infrastructure/zk/src/test/test.ts [hunk 1]
```diff
@@ -4,6 +4,11 @@ import * as utils from '../utils';
 import * as integration from './integration';
 export { integration };
 
+const CODE_COVERAGE_ENABLED = process.env.CODE_COVERAGE;
+const CODE_COVERAGE_FLAGS = `CARGO_INCREMENTAL=0 RUSTFLAGS='-C instrument-coverage' LLVM_PROFILE_FILE='cargo-test-%p-%m.profraw'`;
+
+const codeCoverageFlags = () => (CODE_COVERAGE_ENABLED ? CODE_COVERAGE_FLAGS : '');
```
Comment (jurajpiar) [inline infrastructure/zk/src/test/test.ts:10]: not extremely important, but why is this a function if it only deals with constants?
wouldn't this do?
```ts
const CODE_COVERAGE_FLAGS = process.env.CODE_COVERAGE
    ? `CARGO_INCREMENTAL=0 RUSTFLAGS='-C instrument-coverage' LLVM_PROFILE_FILE='cargo-test-%p-%m.profraw'`
    : '';
```

File: docs/development.md [hunk 2]
```diff
@@ -181,6 +181,33 @@ zk test db
...
+### Code coverage
+
+**Note**. Code coverage measurement requires `grcov` and `llvm-tools-preview` to be installed.
+
+```bash
+cargo install grcov
+rustup component add llvm-tools-preview
+```
+
+To measure code coverage of unit tests, just set the environment variable `CODE_COVERAGE` to `true`.
```
Comment (jurajpiar) [inline docs/development.md:193]: not true strictly speaking as setting it to true should result in `'true'` not `true` and thus `CODE_COVERAGE=false zk test prover` will also work (as `'false'` is truthy) afaik.

File: infrastructure/zk/src/test/test.ts (hunk 1 above)
Comment (jurajpiar) [reply infrastructure/zk/src/test/test.ts:10]: https://github.com/rsksmart/rif-rollup/pull/94/files#r1324815783 ... unless we do `JSON.parse(process.env.CODE_COVERAGE)` ;)

**Review by jurajpiar — DISMISSED** [review]

File: .gitignore [hunk 3]
```diff
@@ -60,3 +60,6 @@ flamegraph.svg
 bin/mkcert
 
 docker/rskj/data
+
+# Files used to generate code-coverage reports
+*.profraw
```
Comment (jurajpiar) [inline .gitignore:65]: add space

File: docs/development.md (hunk 2 above)
Comment (antomor) [reply docs/development.md:193]: Absolutely true, thank you for highlighting it

File: infrastructure/zk/src/test/test.ts (hunk 1 above)
Comment (antomor) [reply infrastructure/zk/src/test/test.ts:10]: I'd prefer to explicitly check that the value specified is either `true` or `1` and applying the flags only in these cases.

File: .gitignore (hunk 3 above)
Comment (antomor) [reply .gitignore]: Done

File: docs/development.md (hunk 2 above)
Comment (antomor) [reply docs/development.md]: Done

File: infrastructure/zk/src/test/test.ts (hunk 1 above)
Comment (antomor) [reply infrastructure/zk/src/test/test.ts]: Done

**Review by jurajpiar — DISMISSED** [review]

**Review by jurajpiar — DISMISSED** [review]

**Review by jurajpiar — DISMISSED** [review]

**Review by jurajpiar — APPROVED** [review]

**Pull request merged**